src/psyclone/psyir/nodes/routine.py

```diff
@@ -212,12 +212,20 @@ def name(self, new_name):
                     f"Can't assign '{new_name}' as the routine name because "
                     f"its symbol table contains a symbol ({existing_symbol}) "
                     f"already tagged as 'own_routine_symbol'.")
+            rsymbol = None
```

Comment:
As mentioned in the discussion, please add a comment here.

Reply:
This will also need a TODO for the new issue you've created about correcting the properties in the other direction.

Reply:
Both comments added.